src/libertem/io/dataset/hdf5.py

```diff
@@ -291,6 +349,133 @@ def _get_tiles_normal(self, tiling_scheme, tileshape_nd, dest_dtype="float32"):
                     scheme_idx=scheme_idx,
                 )
 
+    def _get_reshaped_tiles(self, tiling_scheme, tileshape_nd, dest_dtype="float32"):
+        subslices = self._get_subslices(
+                tiling_scheme=tiling_scheme,
+                tileshape_nd=tileshape_nd,
+            )
+        # reshape tileshape_nd with respect to native shape
+        tileshape_nd = np.unravel_index(
+            np.ravel_multi_index(tuple(i - 1 for i in tileshape_nd), tileshape_nd),
+            self.native_shape
+        )
+        tileshape_nd = tuple(i + 1 for i in tileshape_nd)
+        with self.reader.get_h5ds() as dataset:
+            for scheme_idx, tile_slice in subslices:
+                sig_dims = self.native_shape.sig.dims
+                ds_shape_nav = self.meta.shape[:-sig_dims]
+                native_ds_shape_nav = self.native_shape[:-sig_dims]
+                o_nav = tile_slice.origin[:-sig_dims]
+                o_pos = np.ravel_multi_index(o_nav, ds_shape_nav)
+                s_nav = tile_slice.shape[:-sig_dims]
+                s_nav = tuple(a - 1 for a in s_nav)
+                s_sig = tile_slice.shape[-sig_dims:]
+                axis_size = np.size(dataset, len(native_ds_shape_nav) - 1)
```

Comment:
This means only the last navigation axis is split, right? I think this is fine in practice, but maybe the tile shape should be guaranteed to fit this axis (in `adjust_tileshape`)